producer/kinesis.go

```diff
@@ -123,6 +134,8 @@ func (prod *Kinesis) Configure(conf core.PluginConfig) error {
 
 	prod.streamMap = conf.GetStreamMap("StreamMapping", "default")
 	prod.batch = core.NewMessageBatch(conf.GetInt("BatchMaxMessages", 500))
+	prod.recordMaxMessages = conf.GetInt("RecordMaxMessages", 1)
+	prod.delimiter = []byte(conf.GetString("RecordMessageDelimiter", "\n"))
```

Comment:
I'm currently checking the rest of your code if MaxMessages is switching between old and new, a default value is probably better here (so keep "\n").
As an Idea: I normally try to fix as many errors as possible without the need for a hard error, so maybe it should go like this: 
- if max messages is < 1 set it to 1 and give a warning
- if max message is > 1 and delimiter is "" set it to "\n" and give a warning